### ISSUE TITLE
Hex numbers not recognized as magic numbers

### ISSUE THREAD

**Issue by Reisz** [metrixplusplus/metrixplusplus#45]
Example file:
```c
#include <stdio.h>
#include <stdlib.h>

int main(char argc, char **argv) {
    printf("%s\n", argv[0] + 10);
    printf("%s\n", argv[0] + 0xa);

    return EXIT_SUCCESS;
}
```

Example Run:
```console
$ metrix++ collect --scmn --scmns
[LOG]: WARNING: Logging enabled with INFO level
[LOG]: INFO:    Excluding: ./metrixpp.db
[LOG]: INFO:    Skipping: ./metrixpp.db-journal
[LOG]: INFO:    Processing: ./tmp.c
[LOG]: WARNING: Done (0.13 seconds). Exit code: 0
$ metrix++ limit --max=std.code.magic:numbers:0
[LOG]: WARNING: Logging enabled with INFO level
[LOG]: INFO:    Processing: ./
[LOG]: INFO:    Applying limit: 'std.code.magic:numbers' > 0.0 [applied to 'any' region type(s)]
./tmp.c:4: warning: Metric 'std.code.magic:numbers' for region 'main' exceeds the limit.
        Metric name    : std.code.magic:numbers
        Region name    : main
        Metric value   : 1
        Modified       : None
        Change trend   : None
        Limit          : 0.0
        Suppressed     : False

./:: info: 1 regions exceeded the limit 'std.code.magic:numbers' > 0.0 [applied to 'any' region type(s)]

[LOG]: WARNING: Done (0.02 seconds). Exit code: 1
```

Expected metric value to be `2`, but got `1`.

**Comment by avkonst** [comment]
Hi! Thanks for the report. This would be relatively easy to fix by modifying regexp for the numbers search. The project is community driven, so please submit the PR with the fix and your code sample in unit tests.

**Comment by avkonst** [comment]
Thanks for contribution. Released in 1.7.1
